engine/class_modules/sc_death_knight.cpp

```diff
@@ -2999,6 +2998,38 @@ struct death_knight_action_t : public Base
 
     return base_gcd;
   }
+
+  void execute() override
+  {
+    action_base_t::execute();
+    // If we spend a rune, we have a chance to spread the dot
+    if ( this->triggers_shackle_the_unworthy && p() -> covenant.shackle_the_unworthy -> ok() && p() -> cooldown.shackle_the_unworthy_icd -> is_ready() )
+    {
+      if ( p() -> rng().roll( p() -> covenant.shackle_the_unworthy -> effectN( 5 ).percent() ) )
+      {
+        death_knight_td_t* source = p() -> get_target_data( p()->target );
+        if ( source -> dot.shackle_the_unworthy -> is_ticking() )
```

Comment:
You could group most of these conditions, and switch from get_target_data( p() -> target ) to get_target_data(target)

```
 if ( p() -> covenant.shackle_the_unworthy -> ok() && this->triggers_shackle_the_unworthy && 
      p() -> get_target_data( target ) -> dot.shackle_the_unworthy -> is_ticking() && 
      p() -> cooldown.shackle_the_unworthy_icd -> is_ready() &&
      p() -> rng().roll( p() -> covenant.shackle_the_unworthy -> effectN( 5 ).percent() ) )
    {
```